subiquity/models/installpath.py

```diff
@@ -40,25 +41,48 @@ def paths(self):
                 'maas-password': 'maas-region-controller maas/password password %s' % results['password'],
-                }
+            }
+            self.curtin['late_commands'] = {
+                '900-maas': ['rm', '-f', '/target/etc/maas/rackd.conf'],
+                '901-maas': ['rm', '-f', '/target/etc/maas/region.conf'],
+                '902-maas': ['curtin', 'in-target', '--', 'maas-rack', 'config', '--init'],
+                '903-maas': ['curtin', 'in-target', '--', 'dpkg-reconfigure', '-u', '-fnoninteractive', 'maas-rack-controller'],
+                '904-maas': ['mount', '-o', 'bind', '/proc', '/target/proc'],
+                '905-maas': ['mount', '-o', 'bind', '/sys', '/target/sys'],
+                '906-maas': ['mount', '-o', 'bind', '/dev', '/target/dev'],
+                '907-maas': ['mount', '-o', 'bind', '/target/bin/true', '/target/usr/sbin/invoke-rc.d'],
+                '908-maas': ['chroot', '/target', 'sh', '-c', 'pg_ctlcluster --skip-systemctl-redirect $(/bin/ls /var/lib/postgresql/) main start'],
+                '909-maas': ['chroot', '/target', 'sh', '-c', 'debconf -fnoninteractive -omaas-region-controller /var/lib/dpkg/info/maas-region-controller.config configure'],
+                '910-maas': ['chroot', '/target', 'sh', '-c', 'debconf -fnoninteractive -omaas-region-controller /var/lib/dpkg/info/maas-region-controller.postinst configure'],
+                '911-maas': ['chroot', '/target', 'sh', '-c', 'pg_ctlcluster --skip-systemctl-redirect $(/bin/ls /var/lib/postgresql/) main stop'],
+                '912-maas': ['umount', '/target/usr/sbin/invoke-rc.d'],
+                '913-maas': ['umount', '/target/dev'],
+                '914-maas': ['umount', '/target/sys'],
+                '915-maas': ['umount', '/target/proc'],
+            }
         elif self.path == 'maas_rack':
-            self.packages = {'packages': ['maas-rack-controller']}
-            self.debconf['debconf_selections'] = {
+            self.source = '/media/rack'
+            self.curtin['debconf_selections'] = {
                 'maas-url': 'maas-rack-controller maas-rack-controller/maas-url string %s' % results['url'],
                 'maas-secret': 'maas-rack-controller maas-rack-controller/shared-secret password %s' % results['secret'],
-                }
+            }
+            self.curtin['late_commands'] = {
+                '90-maas': ['rm', '-f', '/target/etc/maas/rackd.conf'],
+                '91-maas': ['curtin', 'in-target', '--', 'maas-rack', 'config', '--init'],
+                # maas-rack-controller is broken, and does db_input & go on the password question in the postinst...
+                # when it should have been done in .config
+                # and it doesn't gracefully handle the case of db_go returning 30 skipped
+                '93-maas': ['curtin', 'in-target', '--', 'sh', '-c', 'debconf -fnoninteractive -omaas-rack-controller /var/lib/dpkg/info/maas-rack-controller.postinst configure || :'],
```

Comment:
Can these be factored out to avoid repetition of the same subset of commands for both the rack and region cases?  Certainly it worries me that we only '|| :' one of the two invocations of /var/lib/dpkg/info/maas-rack-controller.postinst configure.

Reply:
The rack maintainer scripts, take a different a code path, depending on whether or not the region is installed. Thus in *this* case the maintainer scripts failed and shouldn't. This is to be fixed in their maintainer scripts to handle skipped debconf questions correctly.